src/components/multi_select_state_utils.js

```diff
@@ -37,6 +37,50 @@ export const getMinMaxIndexes = (currentIndex, firstItemShiftSelected) =>
     ? { minIndex: currentIndex, maxIndex: firstItemShiftSelected }
     : { minIndex: firstItemShiftSelected, maxIndex: currentIndex };
 
+export const getAvailableIntervalForSelection = (
+  interval,
+  outsideSelectedItems,
+  maxSelectedItems,
+  firstItemShiftSelected
+) => {
+  const { minIndex, maxIndex } = interval;
+  const shouldBeSelect = outsideSelectedItems + (maxIndex - minIndex);
+  return maxSelectedItems > 0 && maxSelectedItems <= shouldBeSelect
+    ? getFixedInterval(
+        interval,
+        outsideSelectedItems,
+        maxSelectedItems,
+        firstItemShiftSelected
+      )
+    : interval;
+};
+
+const getFixedInterval = (
+  { minIndex, maxIndex },
+  outsideSelectedItems,
+  maxSelectedItems,
+  firstItemShiftSelected
+) => ({
+  minIndex:
+    maxIndex === firstItemShiftSelected
+      ? maxIndex - (maxSelectedItems - outsideSelectedItems - 1)
+      : minIndex,
+  maxIndex:
+    minIndex === firstItemShiftSelected
+      ? minIndex + (maxSelectedItems - outsideSelectedItems - 1)
+      : maxIndex
+});
+
+export const getSelectedItemsOutsideInterval = (
+  sourceItems,
+  selectedItems,
+  interval
+) =>
+  selectedItems.filter(selectedItem => {
+    const index = sourceItems.findIndex(item => item.id === selectedItem.id);
+    return !isWithin(index, interval) || selectedItem.disabled;
```

Comment:
why do u need the disable check? can u give me scenario?

Reply:
![Screenshot_5](https://user-images.githubusercontent.com/3428923/62624078-f1eb3780-b92a-11e9-9e7d-67a8b14cfeaf.png)
Sure, please see attached screen